python/GafferUI/NodeSetEditor.py

```diff
@@ -61,22 +77,149 @@ def __init__( self, topLevelWidget, scriptNode, **kw ) :
 		# to avoid being called when they're only half constructed.
 		self.__setNodeSetInternal( self.scriptNode().selection(), callUpdateFromSet=False )
 
-	## Sets the nodes that will be displayed by this editor. As members are
-	# added to and removed from the set, the UI will be updated automatically
-	# to show them. This also calls `nodeSet.setRemoveOrphans( True )` so that
-	# deleted nodes are not visible in the UI.
+	## Sets the nodes that will be displayed by this editor.
+	# As members are added to and removed from the set, the UI will be updated
+	# automatically to show them. This also calls `nodeSet.setRemoveOrphans( True
+	# )` so that deleted nodes are not visible in the UI.
+	#
+	# This will break any editor links, if set.
+	# NOTE: The driver will be updated *after* the node set, such that calling
+	# `getNodeSetDriver` in the nodeSetChangedSignal will return the departing
+	# driver. If the listener requires accurate information about the driver, then
+	# the nodeSetDriverChanegd signal should be used instead.
 	def setNodeSet( self, nodeSet ) :
 
 		self.__setNodeSetInternal( nodeSet, callUpdateFromSet=True )
+		# We do this after setting the node set, so that when the driver changed
+		# signal is emitted, we will have the new node set. Otherwise the editor
+		# looks like it has the old drivers node set still despite not having a
+		# driver...
+		self.setNodeSetDriver( None )
 
 	def getNodeSet( self ) :
 
 		return self.__nodeSet
 
+	## Called before nodeSetDriverChangedSignal in the event that setNodeSet breaks a driver link.
 	def nodeSetChangedSignal( self ) :
 
 		return self.__nodeSetChangedSignal
 
+	## Links the nodeSet for this editor to that of the supplied drivingEditor.
+	# The default mode results in a simple mirroring of the driver's node set
+	# to this editor. Other modes may be registered by other Gaffer modules.
+	# If drivingEditor is None, any existing links will be broken.
+	def setNodeSetDriver( self, drivingEditor, mode = DriverModeNodeSet ) :
+
+		if drivingEditor is not None :
+			assert( isinstance( drivingEditor, GafferUI.NodeSetEditor ) )
+			# We also need to stop people creating infinite loops
+			if GafferUI.NodeSetEditorAlgo.drivesNodeSet( self, drivingEditor ) :
```

Comment:
This seems odd - the general idea with Algo is that it provides functionality above and beyond that of the class itself. To have the class depend on the algo is a bit backwards, no?

Reply:
Totally! Fixed in https://github.com/GafferHQ/gaffer/pull/3270/commits/7548da0d9d0b504dc4e99163cd5bb1197efd54fd